api/turing/config/config.go

```diff
@@ -100,7 +107,58 @@ type DeploymentConfig struct {
 
 // EnsemblingJobConfig captures the config related to the ensembling batch jobs.
 type EnsemblingJobConfig struct {
-	DefaultEnvironment string `validate:"required"`
+	BatchSize          int          `validate:"required"`
+	MaxRetryCount      int          `validate:"required"`
+	DefaultEnvironment string       `validate:"required"`
+	KanikoConfig       KanikoConfig `validate:"required"`
+	ImageConfig        ImageConfig  `validate:"required"`
+	InjectGojekConfig  bool
+}
+
+// ImageConfig provides the configuration used for the OCI image building.
+type ImageConfig struct {
+	LocalRegistry        string
+	Registry             string `validate:"required"`
+	BaseImageRef         string `validate:"required"`
+	BuildNamespace       string `validate:"required"`
+	BuildContextURI      string `validate:"required"`
+	DockerfileFilePath   string `validate:"required"`
+	BuildTimeoutDuration time.Duration
+}
+
+// ResourceRequestsLimits contains the Kubernetes resource request and limits for kaniko
+type ResourceRequestsLimits struct {
+	Requests Resource `validate:"required"`
+	Limits   Resource `validate:"required"`
+}
+
+// Resource contains the Kubernetes resource request and limits
+type Resource struct {
+	CPU    string `validate:"required"`
+	Memory string `validate:"required"`
+}
+
+// KanikoConfig provides the configuration used for the Kaniko image.
+type KanikoConfig struct {
+	Image                  string                 `validate:"required"`
+	ImageVersion           string                 `validate:"required"`
+	ResourceRequestsLimits ResourceRequestsLimits `validate:"required"`
+}
+
+// SparkInfraConfig contains the infra configurations that is unique to the user's Kubernetes
+type SparkInfraConfig struct {
```

Comment:
Nitpick: `SparkInfraConfig` makes me feel Turing is going to set up the Spark infrastructure. Rather, we are only creating the Spark application and taking in some of the config of the existing infra, right? Should we call this `SparkAppConfig` ?

Reply:
Ya I think this is better too.